kytos/core/db.py

```diff
@@ -0,0 +1,139 @@
+"""DB client."""
+# pylint: disable=invalid-name,redefined-outer-name,too-many-arguments
+# pylint: disable=unsubscriptable-object,inconsistent-return-statements
+
+import logging
+import os
+import sys
+import time
+from typing import Optional
+
+import pymongo.helpers
+from pymongo import MongoClient
+from pymongo.errors import AutoReconnect, OperationFailure
+
+from kytos.core.exceptions import KytosDBInitException
+
+LOG = logging.getLogger(__name__)
+
+
+def _log_pymongo_thread_traceback() -> None:
+    """Log pymongo thread traceback.
+
+    Originally, it printed out to sys.stderr poluting it when handling
+    certain asynchronous errors that can happen, with this patched function
+    it logs to LOG.error instead.
+    """
+    if sys.stderr:
+        einfo = sys.exc_info()
+        try:
+            LOG.error(einfo)
+        except IOError:
+            pass
+        finally:
+            del einfo
+
+
+if hasattr(pymongo.helpers, "_handle_exception"):
+    pymongo.helpers._handle_exception = _log_pymongo_thread_traceback
+
+
+def mongo_client(
+    host_seeds=os.environ.get(
+        "MONGO_HOST_SEEDS", "mongo1:27017,mongo2:27018,mongo3:27019"
+    ),
+    username=os.environ.get("MONGO_USERNAME") or "invalid_user",
+    password=os.environ.get("MONGO_PASSWORD") or "invalid_password",
+    database=os.environ.get("MONGO_DBNAME") or "napps",
+    connect=False,
+    retrywrites=True,
+    retryreads=True,
+    readpreference="primaryPreferred",
+    readconcernlevel="majority",
+    maxpoolsize=int(os.environ.get("MONGO_MAX_POOLSIZE") or 300),
+    minpoolsize=int(os.environ.get("MONGO_MIN_POOLSIZE") or 30),
+    serverselectiontimeoutms=30000,
+    **kwargs,
+) -> MongoClient:
+    """Instantiate a MongoClient instance.
+
+    MongoClient is thread-safe and has connection-pooling built in. NApps are
+    supposed to use the Mongo class that wraps a MongoClient. NApps might
+    use a new MongoClient instance for exceptional cases where a NApp
+    needs to parametrize differently.
+    """
+    return MongoClient(
+        host_seeds.split(","),
+        username=username,
+        password=password,
+        connect=connect,
+        authsource=database,
+        retrywrites=retrywrites,
+        retryreads=retryreads,
+        readpreference=readpreference,
+        maxpoolsize=maxpoolsize,
+        minpoolsize=minpoolsize,
+        readconcernlevel=readconcernlevel,
+        serverselectiontimeoutms=serverselectiontimeoutms,
+        **kwargs,
+    )
+
+
+class Mongo:
+    """MongoClient instance for NApps."""
+
+    client = mongo_client(connect=False)
+    db_name = os.environ.get("MONGO_DBNAME") or "napps"
```

Comment:
```suggestion
    db_name = os.environ.get("MONGO_DBNAME", "napps")
```

Reply:
@ajoaoff, initially it was this way. However, after seeing in practice some `pymongo` errors that can happen when a variable is undefined (or non thruthy) I decided to add the `or` fallback to avoid non thruthy values (commit 238d993d). So, as it is we should have clearer and more explicit errors if developers or operators misconfigure env vars.